quartz/plugins/transformers/ofm.ts

```diff
@@ -813,19 +810,6 @@ export const ObsidianFlavoredMarkdown: QuartzTransformerPlugin<Partial<Options>>
         })
       }
 
-      if (opts.mermaid) {
-        js.push({
-          script: mermaidExtensionScript,
-          loadTime: "afterDOMReady",
-          moduleType: "module",
-          contentType: "inline",
-        })
-        css.push({
-          content: mermaidStyle,
-          inline: true,
-        })
-      }
-
```

Comment:
idea for the second step is to add a field to https://github.com/jackyzha0/quartz/blob/7d4bed64a98e0ef982b98e9b755fead7c06855c5/quartz/build.ts#L25 for `contentPresenceChecks` and add a subfield for `hasMermaidDiagram` and then conditionally include the body code

Reply:
Should we include contentPresenceChecks in BuildData or BuildCtx?

For using this downstream in transformers, we prob want to access it via ctx?

Reply:
ah correct, lets do ctx

Reply:
actually i thought about it a bit more, we can do it on a per-file level and have ofm.ts declare a field in vfile like `hasMermaidDiagram` (via the `declare module "vfile"` pattern) and in the mermaid code block parser we set that field to true -- component obv will also have access to that on a per-page basis and we can conditionally exclude the script in content based on that

Reply:
so do you still prefer to have mermaid bundled per file basis, given like you mentioned b4 that mermaid doesn't do treeshake?

Reply:
we could have the script do a dynamic import via url similar to what you had before? i think having a first-load time to render a mermaid diagram is ok and we can have an option to compile-time turn it to svg later

Reply:
sg will do this later tn once I'm back (ooo atm)